Convert doc pages to mkdocs

fyliu opened the pull request:
### Dependency

 - #300

Fixes #167

### What changes did you make?

- use mkdocs formatting where the plain text can be converted (tabs, callouts, dropdown content, etc.)
  - some secondary content are put into dropdowns to simplify the main content of the page
  - some titles are shortened to simplify the page content display on the right. Section titles should not be full sentences anyway.

- fix internal links
- fix md formatting like numbered list
- document common mkdocs formatting syntax

### Why did you make the changes (we will use this info to test)?

- these are the bulk of the remaining issues to fully convert the documentation from plain markdown to mkdocs which helps certain part more clear and stand out better

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->

<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

<details>
<summary>contributing page</summary>

![contributing page](https://github.com/hackforla/peopledepot/assets/1160105/34ad7405-c703-43ec-8f6e-09907f96b581)
</details>
<details>
<summary>add new models and API endpoints page</summary>

![add new models and API endpoints page](https://github.com/hackforla/peopledepot/assets/1160105/e23e5743-98dc-438c-8687-55d77af975bf)
</details>
</details>

<details>
<summary>Visuals after changes are applied</summary>

<details>
<summary>contributing page</summary>

![contributing page](https://github.com/hackforla/peopledepot/assets/1160105/17f80241-6ad0-4d0e-a91c-5457b0e20d9a)
</details>
<details>
<summary>add new models and API endpoints page</summary>

![add new models and API endpoints page](https://github.com/hackforla/peopledepot/assets/1160105/475dc6c2-8c6d-42bb-b010-70ea770ec143)
</details>
<details>
<summary>add new models and API endpoints page expanded</summary>

![add new models and API endpoints page expanded](https://github.com/hackforla/peopledepot/assets/1160105/3acc6bc8-3b27-4051-984c-03fbb144ef52)
</details>
</details>

fyliu commented:
I added examples to the documentation for mkdocs and the description for this PR. This is now ready for review.

freaky4wrld reviewed (changes requested):
@fyliu thanks for making such a vast and detailed documentation..... reviewing it was nice, I came to know across the `Mkdocs` it's useful and UI-friendly.

Here are my observations:
- In the `Contributing.md` after `line 433` you somehow copied the whole document again and it's showing again in the `mkdocs site`. I guess we should remove that duplicity. Below is the reference for you:
<details>
<img width="1440" alt="image" src="https://github.com/hackforla/peopledepot/assets/134206855/6b459b4a-e457-4076-9942-d873ae343a9d">
</details>

- The `add-model-and-api-endpoint.md` everything seems fine, I like the feature where a detailed description is given in the code-block

- The `mkdocs.md` is great as well and you tried to added most of the things.

Making the suggested changes will make it good to go. Thanks for your contribution really appreciate your effort to contribute your knowledge in most laymen terms.

fyliu commented:
@freaky4wrld Thank you so much for catching this duplication. It looks like the page is duplicated even though the page contents on the right doesn't show it.

I guess the duplication has been there for a couple months already. I used `git blame` to see when it last changed.

```
❯ git blame CONTRIBUTING.md -L 433,435
2 months ago    Ethan-Stroming  ec465270│ 433│                                                                        
                                        │ 434│ # Contributing                                                         
                                        │ 435│                                                                        

```

I'll make a ER to resolve any differences within that file before applying the changes from this PR.

fyliu commented on CONTRIBUTING.md:
```diff
@@ -615,10 +626,16 @@ docker-compose down -v
+    1. Delete people-db-1 container
+    1. Open Images Tab
+    1. Remove djangorestapipostrgresql image
 
```
This will be resolved once the PR #301 is merged and these changes are rebased.

fyliu commented on CONTRIBUTING.md:
````diff
@@ -676,7 +693,7 @@ You can probably skip this if you fix the issue on the same day that you pulled
 ```
 
-**Note:** If you are using Visual studios code you can use the Git graphical user interface to stage your changes. For instructions check out the [Git Gui Wiki](<https://github.com/hackforla/website/wiki/How-to-Use-Git-GUI-(Graphical-user-Interface)-in-Visual-Studios-Code>).
+!!! note "If you are using Visual studios code you can use the Git graphical user interface to stage your changes. For instructions check out the [Git GUI page in the website Wiki](<https://github.com/hackforla/website/wiki/How-to-Use-Git-GUI-(Graphical-user-Interface)-in-Visual-Studios-Code>)"
````
This will be resolved once the PR #301 is merged and these changes are rebased.

fyliu commented on CONTRIBUTING.md:
```diff
@@ -742,11 +759,6 @@ git pull upstream
-  <summary><strong>Click here</strong> to see how to sync the fork on GitHub</summary>
-  <img src="https://docs.github.com/assets/cb-49937/images/help/repository/update-branch-button.png" />
-</details>
-
```
This will be resolved once the PR #301 is merged and these changes are rebased.

fyliu commented:
Eliminated duplicate content in CONTRIBUTING.md and updated screenshot.

Also fixed some other problems and added line numbers to code listings.

This is ready for review again.

fyliu commented:
@freaky4wrld can you look at this PR again? Sorry I force pushed. I should have added new fix commits and then do cleanup after the PR is approved.

freaky4wrld reviewed (approved):
@fyliu thanks for making the changes and contributing in creating the documentation visually appealing and easy to understand
- the UI is great 
- there's no duplication this time

PR approved!!
